build-advisories/src/Roave/SecurityAdvisories/VersionConstraint.php

```diff
@@ -142,10 +147,57 @@ public function isUpperBoundIncluded()
      *
      * @return bool
      */
-    public function contains(VersionConstraint $other)
+    public function canMergeWith(VersionConstraint $other)
     {
-        return $this->isSimpleRangeString  // cannot compare - too complex :-(
-            && $other->isSimpleRangeString // cannot compare - too complex :-(
+        return $this->contains($other)
+            || $other->contains($this)
+            || $this->overlapsWith($other)
+            || $other->overlapsWith($this);
+    }
+
+    /**
+     * @param VersionConstraint $other
+     *
+     * @return VersionConstraint
+     *
+     * @throws \LogicException
+     */
+    public function mergeWith(VersionConstraint $other)
+    {
+        if ($this->contains($other)) {
+            return $this;
+        }
+
+        if ($other->contains($this)) {
+            return $other;
+        }
+
+        if ($this->overlapsWith($other)) {
+            return $this->mergeWithOverlapping($other);
+        }
+
+        if ($other->overlapsWith($this)) {
+            return $other->mergeWithOverlapping($this);
+        }
+
+        throw new \LogicException(sprintf(
+            'Cannot merge %s "%s" with %s "%s"',
+            get_class($this),
```

Comment:
you can use ``__CLASS__`` for both class names (or omit it entirely): the class is final, so it can never be a child class.